src/Components/Details/Details.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { FaArrowLeft } from "react-icons/fa";
+import apiKey from "../api/Key";
+import axios from "axios";
+import Card from "../Card/Card";
+import "./details.css";
+
+export default function Details() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [details, setDetals] = useState([]);
+  const [info, setInfo] = useState([]);
+  const [similerResult, setSimilerResult] = useState([]);
+  const [similerCard, setSimilerCard] = useState([]);
+
+  const handleGoBack = () => {
+    navigate(-1);
+  };
+  useEffect(() => {
+    const returnDetails = async (e) => {
+      try {
+        const response = await axios.get(
+          `https://api.spoonacular.com/recipes/${id}/summary?apiKey=${apiKey}`
+        );
+
+        const results = response.data;
+        setDetals(results);
+      } catch (error) {
+        console.error("Error fetching recipes:", error);
+      }
+    };
+    returnDetails();
+  }, [id]);
+
+  useEffect(() => {
+    const returnInfo = async () => {
+      try {
+        const response = await axios.get(
+          `https://api.spoonacular.com/recipes/${id}/information?apiKey=${apiKey}`
+        );
+
+        const { title, image } = response.data;
+        setInfo({ title, image });
+      } catch (error) {
+        console.error("Error fetching recipe information:", error);
+      }
+    };
+
+    returnInfo();
+  }, [id]);
+
+  useEffect(() => {
+    const fetchSimilarRecipes = async () => {
+      try {
+        const response = await axios.get(
+          `https://api.spoonacular.com/recipes/${id}/similar?apiKey=${apiKey}`
+        );
+        const data = response.data;
+        setSimilerResult(data);
+
+        const detailsPromises = data.map(async (similarRecipe) => {
+          const detailsResponse = await axios.get(
+            `https://api.spoonacular.com/recipes/${similarRecipe.id}/information?apiKey=${apiKey}`
+          );
+          return detailsResponse.data;
+        });
+
+        const detailsResults = await Promise.all(detailsPromises);
+        setSimilerCard(detailsResults);
+      } catch (error) {
+        console.error("Error fetching similar recipes:", error);
+      }
+    };
+
+    fetchSimilarRecipes();
+  }, [id]);
```

Comment:
don't create 2 **same** useEffect, if they have different array then will create but here they get the same array [id]
so create only one and call inside it the 2 functions fetchSimilar and returnInfo, you can put the functions outside the useEffect